### ISSUE TITLE
Boxes generate in ocean

### ISSUE THREAD

**Issue by Projecki** [BentoBoxWorld/Boxed#74]
### Expected behavior

Boxes generate not in ocean

### Observed/Actual behavior

boxes generate in ocean sometimes

### Steps/models to reproduce

make a box

### BentoBox version

[18:25:20 INFO]: Running Invalid (UNIVERSESPIGOT) 1.20.4.
[18:25:20 INFO]: (git-UniverseSpigot-"1954a51" (MC: 1.20.4))
[18:25:20 INFO]: BentoBox version: 2.3.0
[18:25:20 INFO]: Database: MARIADB
[18:25:20 INFO]: Loaded Game Worlds:
[18:25:20 INFO]: boxed_world (Boxed): Overworld, Nether
[18:25:20 INFO]: Loaded Addons:
[18:25:20 INFO]: Border 4.3.0 (ENABLED)
[18:25:20 INFO]: Boxed 2.5.0 (ENABLED)

### Other

![Untitled](https://prod-files-secure.s3.us-west-2.amazonaws.com/2db4d6c8-cbae-4740-86bb-7cacf1b92d66/eaed8df2-7af6-4414-a16f-cb0723826dc3/Untitled.png)

![Untitled](https://prod-files-secure.s3.us-west-2.amazonaws.com/2db4d6c8-cbae-4740-86bb-7cacf1b92d66/ef727a24-1d73-44e4-9f60-9f2b93f634e9/Untitled.png)

![Untitled](https://prod-files-secure.s3.us-west-2.amazonaws.com/2db4d6c8-cbae-4740-86bb-7cacf1b92d66/8b243d1d-c896-4be3-be48-75c147244ba6/Untitled.png)

![Untitled](https://prod-files-secure.s3.us-west-2.amazonaws.com/2db4d6c8-cbae-4740-86bb-7cacf1b92d66/e3d068ba-dc27-4aad-bc2f-a8fe09eba747/Untitled.png)

**Comment by Projecki** [comment]
Just released our box realm today, and getting a lot of reports of boxes spawning like the one in the screenshots

**Comment by Projecki** [comment]
![image](https://github.com/BentoBoxWorld/Boxed/assets/65093423/9f1e9670-2e64-44c2-a989-92721133c062)

![image](https://github.com/BentoBoxWorld/Boxed/assets/65093423/7146322c-a295-4f6a-8bc1-0da73fa51b39)

structures generate at least! lol

**Comment by tastybento** [comment]
Do you have multiverse running? Usually when the boxes generate is variable places it's because the world generator isn't working. Multiverse unfortunately isn't compatible with Boxed. It works once but on server reboot it doesn't. 
If you don't have Multiverse do you have something similar?

**Comment by Projecki** [comment]
Yes, I do use multiverse.

I attempted to use MyWorlds but the plugin would not load on my setup. Their dev team wasn’t very helpful

besides, is swapping to my worlds, assuming I can get it working, safe with an already in production server?

**Comment by Projecki** [comment]
I’ve My Worlds loading properly now on my test environment

are there any implications to swapping to My Worlds from Multiverse with my production Boxed server? Or should it be safe to drop it in

**Comment by Projecki** [comment]
https://youtu.be/XdzaJ_dSIdE

Major issues using My Worlds now

MyWorlds 1.20.6-v1-294
BKCommonLib 1.20.6-v1-1749

Boxes generate with no blocks aside from the initial blueprint!

**Comment by tastybento** [comment]
Do you need a multi world plugin? It's not needed just for Boxed.

**Comment by Projecki** [comment]
I do weekly resource world resets, where we use a world manager plugin to delete the resource world and create a new one & then Chunky is scripted to generate the chunks.

I can give no multiverse or my worlds a try, though.

**Comment by tastybento** [comment]
> https://youtu.be/XdzaJ_dSIdE
> 
> 
> 
> Major issues using My Worlds now
> 
> 
> 
> MyWorlds 1.20.6-v1-294
> 
> BKCommonLib 1.20.6-v1-1749
> 
> 
> 
> Boxes generate with no blocks aside from the initial blueprint!
> 
> 

Also, your seed worlds may be messed up due to the Multiverse issue. What happens is that a seed world is created to copy from to the real world. But if the generator was broken then that could be messed up. If MyWorlds works fine on a clean test server then that's the issue. 

Either way, you will need to solve the seed world issue if you don't want to just restart everything. You should run tests but essentially to reset you remove the additional seed world folders inside the main Boxed worlds and let them regenerate.

**Comment by Projecki** [comment]
I see, I didn't understand that it was the initial seed world generation that could be the issue.

Correct, my worlds clean install seems to work fine - though limited testing

I'll try to replicate the issue on my dev environment, and then test deleteing the seed world folders inside the boxed worlds to let them regenerate. Thanks

**Comment by tastybento** [comment]
And just to explain what is happening in the video- the island was made is a spot that previously had been regenerated. The seed world was void at that position so the spot was made void. Then a new island was put in it.

**Comment by tastybento** [comment]
Another less radical remedy in game make islands with your character and then unregister them and make another. Continue until you get to "good" locations that have the expected environment. Then do not delete or purge those other spots. Then when players make new islands they will always be outside of the bad generation area.

**Comment by Projecki** [comment]
to confirm, the folders to delete are:

Boxed_World/seed
Boxed_World/seed_nether

**Comment by Projecki** [comment]
![image](https://github.com/BentoBoxWorld/Boxed/assets/65093423/88d9efa6-1a8e-46e5-8914-b72098bd160e)

I migrated dev to My worlds & deleted the seed folders above

Despite this, the void boxes are still being used

**Comment by Projecki** [comment]
![image](https://github.com/BentoBoxWorld/Boxed/assets/65093423/dabe7aec-156d-4310-bd17-9c75e83e4e6a)

even with no world manager plugin installed at all - BentoBox still uses old bad void boxes

**Comment by Projecki** [comment]
Okay, so if I unregister these void boxes they won't be used anymore.

I need to confirm no new void boxes are being generated. Let me start tracking coordinates

**Comment by Projecki** [comment]
https://youtu.be/DM1XX4IWkxw

I attempted to unregister all "bad boxes"

However, as you can see in this video - boxes generate in a linear path through the world. They go from jungle, to middle of the ocean, to jungle, to middle of the ocean etc.

This is with a world that was initially generated by multiverse, but has had Boxed_world/seed deleted & regenerated without multiverse or my worlds installed on the server

**Comment by Projecki** [comment]
https://youtu.be/xeS1TNHRVOE

Here is a fresh install

I reset my MariaDB
I reset boxed_world

I generated fresh world with only BentoBox, no multiverse nor myworlds installed

Still - boxes generate off-center from the expected spawn point.

**Comment by tastybento** [comment]
So, is this a world different from the one that it generates by default? Are any of the configs changed at all?

**Comment by Projecki** [comment]
No, this is the default BentoBox/Boxed seed, the only things I've changed in config are irrelevant to this i would think:

# of resets,
default protection flags, 

potentially: size of boxes

by default, bentobox has 500x500 boxes I believe

my boxes are smaller: 300x300

**Comment by Projecki** [comment]
https://pastebin.com/nN6eQ5n9

my boxed configuration, for reference

**Comment by Projecki** [comment]
Yo! 

Area-radius needs to be divisible by 16

Using area-radius 320, or 160 both fix issue. My current area radius 150, it breaks.

Not sure how I can fix my currently released server, however...

**Comment by Projecki** [comment]
Around 6,000 boxes have been made, lol

I think we're just gonna tell players to live with it

The void boxes seem to be fixed by switching to my_worlds + resetting seed world - so our remaining issue is ocean boxes.

**Comment by Projecki** [comment]
Going to release a 2nd Boxed server, we need the player slots. On this new server we'll make sure to use radius 160

**Comment by Projecki** [comment]
@tastybento I have done more testing on generating "void boxes" specifically, like in #65 

Using MyWorlds + having an area radius that is not divisible by 16 results in void boxes when boxes that players have reset are reused next server restart by BentoBox.

So MyWorlds & Multiverse both have this issue. 

Additionally, in testing I found that if a server is stopped without MyWorlds installed, then the next restart will not have void boxes, even if MyWorlds is re-installed and active. This implies that the issue is happening when worlds are saved on shut down & a world manager plugin is installed

I'm not sure how helpful this is, since using a bad area-radius number is the nucleus for all my problems.

But I'm not able to reset my server, we have thousands of boxes. So I'm still trying to find a way to fix these issue for my players
